app/controllers/appointments_controller.rb

```diff
@@ -0,0 +1,52 @@
+class AppointmentsController < ApplicationController
+  before_action :set_appointment, only: %i[show update destroy]
+
+  # GET /appointments
+  def index
+    @appointments = current_patient.appointments
+
+    render json: @appointments.to_json(include: { doctor: { only: %i[first_name last_name] } })
+  end
+
+  # GET /appointments/1
+  def show
+    render json: @appointment.to_json(include: { doctor: { only: %i[first_name last_name] } })
+  end
+
+  # POST /appointments
+  def create
+    @appointment = current_patient.appointments.new(appointment_params)
+
+    if @appointment.save
+      render json: @appointment, status: :created, location: @appointment
+    else
+      render json: @appointment.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /appointments/1
+  def update
+    if @appointment.update(appointment_params)
+      render json: @appointment
+    else
+      render json: @appointment.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /appointments/1
+  def destroy
+    @appointment.destroy!
+  end
```

Comment:
- [ ] What happens here if the specialization is not successfully destroyed?  Kindly, add some error handling like you did in the #update method (line 29). 👍